### PR TITLE
Add ADME metabolism from TDC

### PR DESCRIPTION
**Pull request by phalem**
Add metabolism data of:
CYP P450 2C19 Inhibition, Veith et al.
CYP P450 2D6 Inhibition, Veith et al.
CYP P450 3A4 Inhibition, Veith et al.
CYP P450 1A2 Inhibition, Veith et al.
CYP P450 2C9 Inhibition, Veith et al.
CYP2C9 Substrate, Carbon-Mangels et al.
CYP2D6 Substrate, Carbon-Mangels et al.
CYP3A4 Substrate, Carbon-Mangels et al.
from 
https://tdcommons.ai/single_pred_tasks/adme/#metabolism

### CODE REVIEW

**Review by phalem — COMMENTED** [review]
Add benchmark field

File: data/cyp2c9_substrate_carbonmangels/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  units: substrate
+  type: categorical
+  names:
+  - CYP P450 2C9 Substrate
+  - CYP2C9 Substrate
+  - ADME Drug metabolism
+  - Pharmacokinetics metabolism
+  - Substrate toward CYP2C9
+  uris:
+  - https://bioportal.bioontology.org/ontologies/NCIT?p=classes&conceptid=http%3A%2F%2Fncicb.nci.nih.gov%2Fxml%2Fowl%2FEVS%2FThesaurus.owl%23C26633
```
Comment (kjappelbaum) [inline data/cyp2c9_substrate_carbonmangels/meta.yaml:19]: i would really be specific with the URIs and only add them it they really match the target

File: data/cyp2c9_substrate_carbonmangels/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  split_column: split
+identifiers:
+- id: SMILES
+  type: SMILES
+  description: SMILES
+- id: compound_name
+  type: Other
+  names:
+  - drug bank name
+  - drug name pubchem
```
Comment (kjappelbaum) [inline data/cyp2c9_substrate_carbonmangels/meta.yaml:34]: it cannot be both the pubchem and the drug bank name, can it?

File: data/cyp2c9_substrate_carbonmangels/transform.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
+import pandas as pd
+import yaml
+from tdc.single_pred import ADME
+
+def get_and_transform_data():
+    # get raw data
+    target_folder = 'CYP2C9_Substrate_CarbonMangels'
```
Comment (kjappelbaum) [inline data/cyp2c9_substrate_carbonmangels/transform.py:7]: did you actually add the split column to the new dataset?

File: data/cyp2d6_substrate_carbonmangels/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
+name: cyp2d6_substrate_carbonmangels
+description: |-
+  CYP2D6 is primarily expressed in the liver. It is also highly expressed
+  in areas of the central nervous system, including the substantia nigra. TDC
+  used a dataset from Carbon Mangels et al, which merged information on substrates
+  and nonsubstrates from six publications.
+targets:
+- id: CYP2D6_Substrate
+  description: The drugs that are metabolized by the CYP P450 2D6(1) or not (0)
```
Comment (kjappelbaum) [inline data/cyp2d6_substrate_carbonmangels/meta.yaml:9]: how are the drugs identified here? SMILES? Pubchem ID? Drug net id?

**Review by kjappelbaum — CHANGES_REQUESTED** [review]

File: data/cyp2c9_substrate_carbonmangels/transform.py (hunk 3 above)
Comment (phalem) [reply data/cyp2c9_substrate_carbonmangels/transform.py:7]: Sorry, I forget that. I will take a look again on the code of split.

File: data/cyp2d6_substrate_carbonmangels/meta.yaml (hunk 4 above)
Comment (phalem) [reply data/cyp2d6_substrate_carbonmangels/meta.yaml:9]: It has two identifier, SMILES and drug name or pubchem Synonymous. No pubchem id.

Comment (phalem) [reply data/cyp2d6_substrate_carbonmangels/meta.yaml:9]: It has two identifier, SMILES and drug name or pubchem Synonymous. No pubchem id.

**Comment by phalem** [comment]
Add benchmark to transform.py

File: data/cyp_p450_1a2_inhibition_veith_et_al/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    substrates for this enzyme include caffeine, aflatoxin B1, and acetaminophen.
+targets:
+    - id: CYP1A2_inhibition
+      description: ability of the drug to inhibit CYP P450 1A2 (1) or not (0)
+      units:
+      type: boolean
+      names:
+          - noun: CYP P450 1A2 inhibition
+          - noun: CYP 1A2 inhibition
+          - adjective: inhibits CYP P450 1A2
```
Comment (kjappelbaum) [inline data/cyp_p450_1a2_inhibition_veith_et_al/meta.yaml:18]: adjective or verb? :D i have no clue

**Review by kjappelbaum — CHANGES_REQUESTED** [review]
one comment, otherwise lgtm

File: data/cyp_p450_1a2_inhibition_veith_et_al/meta.yaml (hunk 5 above)
Comment (MicPie) [reply data/cyp_p450_1a2_inhibition_veith_et_al/meta.yaml:18]: google search says verb ;-)

Comment (MicPie) [inline data/cyp_p450_1a2_inhibition_veith_et_al/meta.yaml:18]:
```suggestion
          - verb: inhibits CYP P450 1A2
```

**Review by MicPie — CHANGES_REQUESTED** [review]

**Review by MicPie — APPROVED** [review]

**Pull request merged**